src/components/range-inputs.tsx

```diff
@@ -0,0 +1,211 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_range-input.scss';
+import { useState, useRef } from 'react';
+import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
+import InputSwitch from '@isrd-isi-edu/chaise/src/components/input-switch';
+
+const INTEGER_REGEXP = /^\-?\d+$/;
+
+const FLOAT_REGEXP = /^\-?(\d+)?((\.)?\d+)?$/;
+
+const TIMESTAMP_FORMAT = 'YYYY-MM-DDTHH:mm';
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+/**
+ * an object for mapping the error types to error messages
+ */
+const errorMsgMap: {
+    [key: string]: string;
+} = {
+    'empty': 'Please enter either From or To value',
+    'range': 'From value cannot be greater than the To value',
+    'int': 'Please enter an integer value',
+    'float': 'Please enter a decimal value',
+    'date': 'Please enter a date value',
+    'timestamp': 'Please enter a valid date and time value'
+};
+
+type RangeInputProps = {
+    /**
+     * an string for the type of input expected:
+     * int
+     * float
+     * date
+     * timestamp
+     */
+    inputType: string,
+    /**
+     * use for applying custom styles on the component
+     */
+    classes?: string
+};
+
+const getType = (inputType: string): string => {
+
+    if (inputType.indexOf('int') > -1) return 'int';
+
+    if (inputType.indexOf('float') > -1) return 'float';
+
+    if (inputType.indexOf('date') > -1) return 'date';
+
+    if (inputType.indexOf('timestamp') > -1) return 'timestamp';
+
+    return 'Invalid_Type';
+}
+
+const RangeInput = ({ inputType, classes }: RangeInputProps) => {
```

Comment:
This is a container for both to and from inputs, the name should be plural.
`RangeInputs`

Reply:
done